pkg/reconciler/eventbus/controller.go

```diff
@@ -57,15 +57,54 @@ func (r *eventBusReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	if reconcileErr != nil {
 		log.Errorw("reconcile error", zap.Error(reconcileErr))
 	}
-	if r.needsUpdate(eventBus, busCopy) {
-		// Use a DeepCopy to update, because it will be mutated afterwards, with empty Status.
-		if err := r.client.Update(ctx, busCopy.DeepCopy()); err != nil {
-			return reconcile.Result{}, err
-		}
+
+	// Update the finalizers
+	// We need to always do this to ensure that the field ownership is set correctly,
+	// during the migration from client-side to server-side apply. Otherewise, users
+	// may end up in a state where the finalizer cannot be removed automatically.
+	patch := &v1alpha1.EventBus{
+		Status: busCopy.Status,
```

Comment:
Do we need to add `Status` (even though adding it won't make any difference I think)? Since this is only supposed to update the finalizers.